### ISSUE TITLE
Raise a error if web-handler returned non-response in debug mode.

### ISSUE THREAD

**Issue by benitogf** [aio-libs/aiohttp#3311]
## Long story short

Opening this as mentioned here: https://github.com/aio-libs/aiohttp/issues/3223#issuecomment-426154789

```
Unhandled exception
Traceback (most recent call last):
  File "/tmp/tmpg2n5yr0j/pypi__aiohttp_3_4_4/aiohttp/web_protocol.py", line 410, in start
    await resp.prepare(request)
AttributeError: 'NoneType' object has no attribute 'prepare'
```

## Expected behaviour

It should't trow

## Actual behaviour

There's an unhandled exception thrown

## Steps to reproduce

This happens in a websocket route handler when a client disconnects

## Your environment

aiohttp 3.4.4
OS centos
no proxy

**Comment by yogeshsajanikar** [comment]
I have same issue with streaming response.

**Comment by asvetlov** [comment]
Guys, looks like you both don't return a response object from failed web-handler.

**Comment by benitogf** [comment]
Thanks @asvetlov I was missing the return statement at the end of the handler https://aiohttp.readthedocs.io/en/stable/web_quickstart.html#websockets

**Comment by asvetlov** [comment]
That's exactly what I was suspected.
Runtime check for the return value is quite expensive but we can do it in debug mode (`loop.get_debug()`).

The patch is trivial, a champion is needed.

**Comment by benitogf** [comment]
@asvetlov please review above patch, I saw no use of `loop.get_debug()` in web_protocol so used what was around

**Comment by asvetlov** [comment]
The idea is dropping custom debug checks everywhere in the server and rely on `loop.get_debug()` only.
The change is scheduled to aiohttp 4.0, so `self.debug` check is correct now.

**Comment by asvetlov** [comment]
Fixed by #3321
